### DISCUSSION TITLE
Implementation Question

### DISCUSSION THREAD

**Discussion by sjoram** [alienatedsec/solis-ha-modbus-cloud#5] [Q&A]
Hi - not so much an issue but some questions and I'm not that familiar with Github so unsure of where to ask them!
I received the Waveshare modules today and am in the process of getting things setup.

I have both Server and Client on my network but my first question relates to the Device and Destination Port numbers. The default on the units seems to be 4196. Do I need to change this to 502 or some other port or does it matter as long as they talk to each other?

Second question: I haven't connected the RS485 sides yet but does the attached VirCom look OK?

Lastly, I guess I could confirm the above myself but I'm afraid to pull the datalogger from the inverter! I'm thinking power down - unscrew logger - connect TCP Server - power up and see if I can poll some registers. If I come up against issues then power down again - reinsert the logger - power up and try again another day. I guess my question is will the logger hold all its settings and just power up properly or will I have to set it up again? I would have no idea how to do the latter hence my reticence in disconnecting it :(

Any comments or tips appreciated at this stage.

![VirCom1](https://user-images.githubusercontent.com/121193063/226689004-1c549a3f-59a5-4e82-8b2d-c8e8f22ec5cd.JPG)

**Comment by alienatedsec** [comment]
`TCP Client` will need the `Destination Port` of `TCP Server` See mine highlighted below - yours will need to be set to `4196`. I also use the same port on TCP Server for HA communication.

![image](https://user-images.githubusercontent.com/73167064/226705321-57336f55-4240-4c63-9d3b-6e3eac91aadb.png)

Regarding the logger - I unplugged mine and plugged in the homemade cable with no issues. Maybe, it is better not to make the change in full-blown PV generation or under high load. No need to power the inverter off.

My logger `DLS-W` and `S2-WL-ST` both kept the settings, so there is no harm in unplugging those too.

**Comment by sjoram** [comment]
OK! Many thanks for updating your readme!

I have half of it working! i.e. I have HA and the SolaX integration pulling data from the inverter. Unfortunately I could not get a male plug to fit the datalogger so that is for another day.

I had a problem earlier where the integration went offline. I spent two hours trying to get it back with numerous re installs to no avail. Eventually restarted the Waveshare device with vircom and hey presto it all came back to life. I'm wondering if some timeouts or blocks are still too big and locking it up.

Any ideas welcome!

**Reply by alienatedsec** [reply]
@sjoram great to hear that the HA is working for you. The settings are optimised for `DLS-W`, so you will need to experiment with your setup if a different datalogger or network components.

If you don't have a data logger, the `auto-query storage` settings are not necessary as only one device (that is HA) is connected to the `TCP Server`

**Comment by sjoram** [comment]
@alienatedsec Well it is kind of working :( I have a data logger but not connected back in yet due to the lack of a connector. I hope to rectify that today! Mine is the S3-WIFI one so 'hopefully' it will work.

My problem now is that the HA SolaX Modbus integration keeps disconnecting after a few hours and not reconnecting. The only way I have found to get it going again is to restart the Waveshare server device using VirCon and then reload the integration. Just restarting HA does not get it going.

Any ideas where to start looking for a solution to this by any chance?

**Reply by alienatedsec** [reply]
@sjoram try to swap the units and reconfigure to see if the same will happen. In the other discussion #6 we found not a fully working unit.

**Comment by sjoram** [comment]
Hi - yes I will swap the units but to be honest I don't think that might be the issue. It worked perfectly all day yesterday and had to be restarted twice today. It would be great to be able to see some error logs. Also I really don't need all 100+ registers on a fast poll basis - just 3 or 4. I would suspect that system is hanging with too many registers and too fast timing. It would be good to be able to just poll a few registers and get the rest of them through the logger.

**Reply by alienatedsec** [reply]
@sjoram The idea is not to pull more often than every 5 seconds, but some inverters may not be as efficient and the number of seconds would need to be increased e.g. to 15 seconds. It doesn't matter how many registers you read - 5 or 10 or 100, those are read in bulks.

**Reply by sjoram** [reply]
@alienatedsec Hi - back to this again, I don't pull anywhere near 5 seconds. I think it does matter how many registers you read as RS485 is a serial link. The bulk read is also only effective if the registers are in contiguous memory ranges. It may be that the thing pulls a lot of empty space between registers so wasting time. I really only need 4 or 5 fast values. For the rest I am happy with the cloud values so it would be good to just be able to tag those that I require rather than pulling all 100 every time.

On a second note I have now connected the datalogger up to the TCP/IP slave and can confirm it does 'sometimes' update! It has been working for the last 20 hours but suddenly stopped again about an hour ago.

The HA - Waveshare - Solis channel is also locking up randomly. Sometimes I get more than 24hours and other times not so much. If I restart the Server device using VirCom that 'sometimes' fixes it but usually I also have to reload the Solax integration to trigger it all off again.

I'd really like to be able to review some transmission logs either / and from the Waveshare or the Solax Integration but I'm clueless as to where to start looking for those at the moment!

**Reply by alienatedsec** [reply]
@sjoram The baud rate is 9600 (1200 Bytes per second) which should be more than enough to pull 150 registers at a time. 
You have 80 Bytes per 40 registers ([explained here](https://github.com/wills106/homeassistant-solax-modbus/issues/340#issuecomment-1464112568)), so around 300 Bytes for 150 registers (if so). There must be something wrong with your setup or you don't run the latest beta version to mitigate those issues.

https://github.com/wills106/homeassistant-solax-modbus/releases

![image](https://user-images.githubusercontent.com/73167064/229121211-9b74ec50-9869-4e5d-bf95-d33041edf8a4.png)
